src/dotnet/CarbonAware.DataSources.Json/JsonDataSource.cs

```diff
@@ -67,9 +68,15 @@ private IEnumerable<EmissionsData> FilterByDateRange(IEnumerable<EmissionsData>
     {
         DateTimeOffset end;
         DateTimeOffset.TryParse(endDate.ToString(), out end);
-        data = data.Where(ed => ed.TimeBetween(startDate, end));  
-
-        return data;
+        var (newStartTime, newEndTime) = IntervalHelper.ShiftDate(startDate, end, MinSamplingWindow);
```

Comment:
NIT: I'd prefer more naming consistency.  `newStartTime`, `startDate`, `end` is a little confusing given that they are all `DateTimeOffsets`.  Personal preference for  `newStartTime`, `startTime`, `endTime`, but anything consistent would be great